File: libs/dotnet/SquareSums/NativeNodesSorting.cs
using System;

// ReSharper disable ForCanBeConvertedToForeach

namespace SquareSums
{
    public class NativeNodesSorting: INodesSorting
    {
        private readonly NodesComparer _comparer;

        public NativeNodesSorting(Path? path, int maxN)
        {
            _comparer = new NodesComparer(path, maxN);
        }
        
        public void SortNodes(Span<Node> nodes)
        {
            _comparer.FlushCache();
            
            nodes.Sort(_comparer);
        }
    }
}

File: libs/js/src/NativeNodesSorting.ts
import { Path } from './Path';
import { Node } from './Node';
import { NodesComparer } from './NodesComparer';


export class NativeNodesSorting {
    private _comparer: NodesComparer;

    constructor(path: Path | null, maxN: number) {
        this._comparer = new NodesComparer(path, maxN);
    }

    public sortNodes(nodes: Array<Node>) {
        this._comparer.flushCache();
        nodes.sort((a, b) => this._comparer.compare(a, b));
    }
}